src/convert.js

```diff
@@ -45,65 +43,68 @@ function main() {
   }
 }
 
-function getTokensByCategory(tokensData) {
-  return Object.keys(tokensData.categories).map(key => {
+function getTokensByCategory(tokensData, tokens) {
+  return Object.entries(tokensData.categories).map(([key, category]) => {
     return {
       category: key,
-      prefix: tokensData.categories[key].tokenPrefix,
-      tokens: []
+      prefix: category.tokenPrefix,
+      tokens: getCategoryTokens(key, category.tokenPrefix, tokens)
     };
   });
 }
 
-function resolveCategoryDataForTokens(tokensByCategory, tokens) {
-  return tokensByCategory.map(category => {
-    Object.keys(tokens).forEach(key => {
-      if (tokens[key].category === category.category) {
-        tokens[key].name = key;
-        tokens[key].prefix = `${category.prefix}`;
-        category.tokens.push(tokens[key]);
-      }
-    });
-    return category;
-  });
+function getCategoryTokens(name, prefix, tokens) {
+  return Object.entries(tokens).reduce((resultArray, [key, token]) => {
+    if (token.category === name) {
+      resultArray.push({
+        ...token,
+        name: key,
+        prefix: `${prefix}`
+      });
+    }
+    return resultArray;
+  }, []);
 }
 
-function exportToScss(resolvedTokensByCategory, outDir, outFileName) {
-  const scssExportData = formatToScss(resolvedTokensByCategory);
-  exportFile(`${outDir}`, `${outFileName}.scss`, scssExportData);
+function exportToScss(tokensByCategory, scssPrefix, outDir, outFileName) {
+  const scssExportData = formatToScss(tokensByCategory, scssPrefix);
+  exportFile(`${outDir}`, `${outFileName}.scss`, scssExportData.join(""));
 }
 
-function formatToScss(resolvedTokensByCategory) {
-  let scssExport = [];
-  resolvedTokensByCategory.forEach(category => {
+function formatToScss(tokensByCategory, scssPrefix) {
+  return tokensByCategory.reduce((resultArray, category) => {
     switch (category.category) {
       case "colors":
-        scssExport.push(...formatColorsToScss(category.tokens));
+        resultArray.push(...formatColorsToScss(category.tokens, scssPrefix));
         break;
       case "typography":
-        scssExport.push(...formatTypographyToScss(category.tokens));
+        resultArray.push(
+          ...formatTypographyToScss(category.tokens, scssPrefix)
+        );
         break;
       case "spacing":
-        scssExport.push(...formatSpacingToScss(category.tokens));
+        resultArray.push(...formatSpacingToScss(category.tokens, scssPrefix));
         break;
       default:
         console.warn("Unrecognized category type");
```

Comment:
This is fine, I prefer error first coding practices - not so much difference when functionality is simple.